api/externalLinks/ExternalLinksRestHandler.go

```diff
@@ -0,0 +1,218 @@
+	userService user.UserService,
+	enforcer casbin.Enforcer,
+	validator *validator.Validate, userAuthService user.UserAuthService,
+	deleteService delete2.DeleteService,
+) *ExternalLinksRestHandlerImpl {
+	return &ExternalLinksRestHandlerImpl{
+		logger:               logger,
+		externalLinksService: externalLinksService,
+		userService:          userService,
+		validator:            validator,
+		enforcer:             enforcer,
+		userAuthService:      userAuthService,
+		deleteService:        deleteService,
+	}
+}
+func (impl ExternalLinksRestHandlerImpl) CreateExternalLinks(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	var beans []*externalLinks.ExternalLinksRequest
+	err = decoder.Decode(&beans)
+	if err != nil {
+		impl.logger.Errorw("request err, SaveLink", "err", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	token := r.Header.Get("token")
+	if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionCreate, "*"); !ok {
+		common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+		return
+	}
+
+	res, err := impl.externalLinksService.Create(beans, userId)
+	if err != nil {
+		impl.logger.Errorw("service err, SaveLink", "err", err, "payload", beans)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+func (impl ExternalLinksRestHandlerImpl) GetExternalLinksTools(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+
+	// auth free api as we using this for multiple places
+	/*
+		token := r.Header.Get("token")
+		if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionGet, "*"); !ok {
+			common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+			return
+		}
+	*/
+
+	res, err := impl.externalLinksService.GetAllActiveTools()
+	if err != nil {
+		impl.logger.Errorw("service err, GetAllActiveTools", "err", err)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	common.WriteJsonResp(w, err, res, http.StatusOK)
+}
+func (impl ExternalLinksRestHandlerImpl) GetExternalLinks(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	v := r.URL.Query()
+	id := v.Get("clusterId")
+	clusterId := 0
+	if len(id) > 0 {
+		clusterId, err = strconv.Atoi(id)
+		if err != nil {
+			impl.logger.Errorw("request err, FetchOne", "err", err, "id", id)
```

Comment:
logging is not correct

Reply:
comment msg i guess incorrect here, fixed